pkg/aws/policies.go

```diff
@@ -479,55 +479,69 @@ func (c *awsClient) FindRoleARNs(roleType string, version string) ([]string, err
 		if !strings.Contains(aws.StringValue(role.RoleName), AccountRoles[roleType].Name) {
 			continue
 		}
-		listRoleTagsOutput, err := c.iamClient.ListRoleTags(&iam.ListRoleTagsInput{
-			RoleName: role.RoleName,
-		})
+		isValid, err := c.ValidateAccountRoleMinVersionTag(*role.RoleName, roleType, version)
 		if err != nil {
 			return roleARNs, err
 		}
-		skip := false
-		isTagged := false
-		for _, tag := range listRoleTagsOutput.Tags {
-			tagValue := aws.StringValue(tag.Value)
-			switch aws.StringValue(tag.Key) {
-			case tags.RoleType:
-				isTagged = true
-				if tagValue != roleType {
-					skip = true
-					break
-				}
-			case tags.OpenShiftVersion:
-				isTagged = true
+		if !isValid {
+			continue
+		}
+		roleARNs = append(roleARNs, aws.StringValue(role.Arn))
+	}
+	return roleARNs, nil
+}
 
-				if roleHasTag(listRoleTagsOutput.Tags, tags.ManagedPolicies, tags.True) {
-					// Managed policies will be up-to-date no need to check version tags
-					break
-				}
+// FIXME: refactor similar calls to use this instead
+func (c *awsClient) ValidateAccountRoleMinVersionTag(
```

Comment:
This function also fetches the tags and checks the role type. I guess you extracted the code to reuse the function. But I wonder if we should change the function name or extract only the part under `case tags.OpenShiftVersion:` Which actually checks the minmum version, WDYT?

Reply:
I see, `ValidateAccountRoleComptability` or `ValidateAccountRoleVersionComptability`?